analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -92,49 +101,18 @@ public void ThrowInLoop(object o)
 
         public void Foo1(bool a, bool b)
         {
-            var x = t || a || b; // Noncompliant
-//                  ^
+            var x = t || a || b; // Compliant t is const
         }
 
         public void Foo2(bool a, bool b)
         {
-            var x = ((t)) || a || b; // Noncompliant
-//                    ^
+            var x = ((t)) || a || b; // Compliant t is const
 
         }
 
-        public void Foo3(bool a, bool b)
-        {
-            var x = ((t || a)) || b;
-//                    ^
-        }
-
         public void Foo4(bool a, bool b)
         {
-            var x = ((f || t)) || a || b;
-//                         ^ Noncompliant
-//                    ^ Noncompliant@-1
-
-        }
-
-        public void Foo5(bool a, bool b)
-        {
-            var x = ((f && t)) || a || b;
-//                    ^ Noncompliant
-        }
-
-        public void Foo6(bool a, bool b)
-        {
-            var x = t || a ? a : b;
-//                  ^ Noncompliant
-        }
-
-        public void Foo7(bool a, bool b)
-        {
-            if ((t || a ? a : b) || b)
-//               ^ Noncompliant
-            {
-            }
```

Comment:
I deleted most of there  - not sure what's the point since we don't raise for constants.

Reply:
I'd say change them to `readonly` instead.

Reply:
since we don't track the values of read-only fields - does it offer something :/ 
Maybe adding one to document the behavior - but all of them might be a bit too much. WDTY?

Reply:
Let's change them to local variables then. I think these tests are valuable.